### ISSUE TITLE
VBL & 2x / full speed mode

### ISSUE THREAD

**Issue by burniouf** [AppleWin/AppleWin#1043]
Is there really no way to get VBL still at 50 or 60 Hz while in accelerated CPU speed ?

**Comment by sicklittlemonkey** [comment]
Currently it's calculated from the same clock cycles as the CPU.

What is your user case and suggested alternate to the current behaviour?

**Comment by burniouf** [comment]
keeping VBL (and probably other signals ?) disconnected from CPU speed would make behavior much more like if you have an accelerator board. I'm working on A2osX, and VBL/IRQ is used as system clock for slicing, and for many TCP/IP expiration and other timeout....

**Comment by tomcw** [comment]
Also see #588

**Comment by burniouf** [comment]
Agree, providing both "acceleration" modes could be a good idea....

**Comment by sicklittlemonkey** [comment]
Because AppleWin only supports //e VBL detection, the implementation might
be quite simple.

Just use a wall time like system time, ticks, or high resolution timer to
flag VBL instead of the clock algorithm.

**Comment by burniouf** [comment]
so both polling VBL IO at $C019 & enabling VBL IRQ thru MOUSE firmware would be 50/60hz regardless CPU speed ?

**Comment by sicklittlemonkey** [comment]
Ah, good point. I've never owned a mouse card, and never looked into its
emulation.
Just had a look, and it won't work well for that because it sets up an
event based on clock cycles.
I suppose there could be another event queue based on wall time, but Tom's
not gonna like that idea. : - D

**Comment by burniouf** [comment]
wouldn't it be possible to let actual slider behavior unchanged, and add a secondary slider to speed up CPU only with additional checkboxes for each slot to "slowdown to 1 Mhz" like most acc boards ?

**Comment by tomcw** [comment]
The throttled case, eg. CPU accelerated to 2MHz or 4MHz (or some other fix target like 2.8MHz) is easy I think, as VBl just takes a multiple (eg. x2, x4, x2.8) of CPU cycles now.

In fact VBl is derived from the video scanner. So that would have the same CPU cycle multiplier. 

I haven't considered the unthrottled (ie. Full speed) case though. Perhaps it just needs to be approximately right.
